File: src/main/java/ar/edu/um/haberes/rest/model/CargoClasePeriodo.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CargoClasePeriodo extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 3290045988457434072L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cargoClasePeriodoId;

	private Long legajoId;
	private Long cargoClaseId;
	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Long periodoDesde;
	private Long periodoHasta;
	private Integer horas = 0;
	private BigDecimal valorHora = BigDecimal.ZERO;
	private String descripcion;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "cargoClaseId", insertable = false, updatable = false)
	private CargoClase cargoClase;

	@OneToOne
	@JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
	private Dependencia dependencia;

	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;

}

File: src/main/java/um/haberes/rest/kotlin/model/CargoClasePeriodo.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table
data class CargoClasePeriodo(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoClasePeriodoId: Long? = null,

    var legajoId: Long? = null,
    var cargoClaseId: Long? = null,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var periodoDesde: Long? = null,
    var periodoHasta: Long? = null,
    var horas: Int = 0,
    var valorHora: BigDecimal = BigDecimal.ZERO,
    var descripcion: String? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "cargoClaseId", insertable = false, updatable = false)
    var cargoClase: CargoClase? = null,

    @OneToOne
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: Facultad? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: Geografica? = null

) : Auditable()
